### PR TITLE
Frame pointer tests: stop backtraces at caml_main/caml_startup

### PR DESCRIPTION
**Pull request by gasche**
I read #12587 this morning and I decided to do something about it: the result is the present PR.

What the PR does is to tweak the C function that prints the backtrace to stop at `caml_startup` or `caml_main`. 

In the meantime people pointed to #11594, an existing PR solving the same problem. But that PR seems stuck in a discussion on which scripting language is best to post-process the output, while the present proposal conveniently uses C to process the output. It could have better chances of being integrated in finite time.
(cc @fabbing who would be an ideal reviewer)
**Edit**: in fact it looks like #11594 eventually also decided to use C as its scripting language. The main difference is that the present PR only implements the "stop at caml_main" part in C, and leaves the existing sed script unchanged, while #11594 implements all the logic in C, which makes for a much larger change.

Note: initially I would stop as `caml_start_program`, but in fact this does not quite work because some backtraces include a "fake" `caml_start_program` entry followed by a `caml_callback` entry. I think that comes from the fact that the `caml_callback_asm` implementation jumps to a label inside `caml_start_program`. Trying to detect these "fake" entries in C code reading function calls line by line is just too cumbersome, so instead I stop at caml_start/caml_main.

### CODE REVIEW

File: testsuite/tests/frame-pointers/fp_backtrace.c
Comment (ghost) [inline testsuite/tests/frame-pointers/fp_backtrace.c:117]: You are leaking `symbol` here. It needs to be `free`d prior to `return`.

Comment (gasche) [reply testsuite/tests/frame-pointers/fp_backtrace.c:117]: Indeed. Fixed, thanks!

**Review by ghost — APPROVED** [review]

**Comment by gasche** [comment]
Thanks! This would need a maintainer on-behalf-of approval before merging. ( cc @Octachron ? )

**Comment by shindere** [comment]
Gabriel Scherer (2023/10/02 07:44 -0700):
> Thanks! This would need a maintainer on-behalf-of approval before
> merging. ( cc @Octachron ? )

This PR has been discussed during the last triaging meeting. During this
meeting, it has been decided that @damiendoligez will look at this PR
and compare its merits to the earlier #11594. I thus think this PR
should not be merged before @damiendoligez had a chance to look at both
this one and #11594.

**Comment by gasche** [comment]
I disagree.

**Comment by shindere** [comment]
Gabriel Scherer (2023/10/04 00:54 -0700):
> Merged #12588 into trunk.

You yourself wrote:

```
Thanks! This would need a maintainer on-behalf-of approval before
merging. ( cc @Octachron ? )
```

**Comment by gasche** [comment]
~~Yes, but I changed my mind because of your annoying stalling tactics; I don't want to have to ask someone else to override you, so I decided to go ahead.~~ (**Edit**: this was written in anger, [see below](https://github.com/ocaml/ocaml/pull/12588#issuecomment-1751733816) for context.)

I understand that you appreciate the work of @fabbing, and I also do, but it is obvious that the present PR is a good first step on top of which #11594 could be rebuilt if we wanted to, and not an equal alternative to compare in length. The PR has been reviewed and it was good to merge -- modulo the drama.

**Comment by Octachron** [comment]
I am reverting this as a breach of our reviewing process. Please avoid merging PRs on negative emotional highs.

**Comment by gasche** [comment]
My previous reaction was written in anger, and (as pointed out by @Octachron above) this is not a good behavior. Here is a calmer exposition of how I got in this situation:

- I consider the behavior of @shindere in this PR to be obstructive rather than constructive. At no point did he offer help on the original issue or on reviewing / making decision on the change proposed.

- @damiendoligez is one of the maintainers with by far the highest latency (counted in weeks), and asking someone to wait on his actions is the github/ocaml equivalent of "please wait until pigs fly".

**Comment by damiendoligez** [comment]
For the record, @gasche, @shindere, @Octachron and I had a short in-person meeting and managed to settle our difference of point of view.

**Pull request merged**
